src/containers/pages/JurisdictionAssignment/helpers/SelectedJurisdictionsCount/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Fragment, useState } from 'react';
+import { Tooltip } from 'reactstrap';
+import { TreeNode } from '../../../../../store/ducks/opensrp/hierarchies/types';
+
+/** Props for SelectedJurisdictionsCount  */
+export interface SelectedJurisdictionsCountProps {
+  id: string;
+  jurisdictions: TreeNode[] /** array of jurisdictions */;
+  parentNode: TreeNode | undefined;
+}
+
+/** default props for SelectedJurisdictionsCount */
+const defaultProps: SelectedJurisdictionsCountProps = {
+  id: '',
+  jurisdictions: [],
+  parentNode: undefined,
+};
+
+/**
+ * getSelectedNodesUnderParentNode
+ *
+ * Disects the selected nodes to the respective children as you drill down the table
+ *
+ * @param tree - Parent node of the tree
+ * @param selectedLeafNodes - an array of all the selected leaf nodes
+ */
+const getSelectedNodesUnderParentNode = (
+  tree: TreeNode | undefined,
+  selectedLeafNodes: TreeNode[]
+): TreeNode[] => {
+  const nodesList: TreeNode[] = [];
+  if (!tree) {
+    return selectedLeafNodes;
+  }
+  tree.walk(node => {
+    selectedLeafNodes.forEach(leaf => {
+      if (leaf.model.id === node.model.id) {
+        nodesList.push(leaf);
+      }
+    });
+    return true;
+  });
+  return nodesList;
+};
```

Comment:
Is it possible to replace this with the selector? i.e. https://github.com/onaio/reveal-frontend/blob/master/src/store/ducks/opensrp/hierarchies/index.ts#L331

Reply:
I'm not sure it would work... i'm using https://github.com/onaio/reveal-frontend/blob/master/src/store/ducks/opensrp/hierarchies/index.ts#L331 and filtering out the items depending on the e drill down level

Reply:
in that case then this function needs to be a selector function in the redux module no?  My reasoning is that 1) the 
purpose of selectors is to get/derive/calculate data from the store and 2) if we have a [memoized selector](https://github.com/reduxjs/reselect) then we improve performance - we might need that because your function here is traversing the whole tree for each parent's descendants.